code/dashboard-2/src/util/query/QueryUtils.ts

```diff
@@ -354,6 +357,71 @@ export function makeFilter(query: string) {
   }
 }
 
+export const prepareJobQueryString = (jobQueryValues?: JobQueryValues) => {
+  if(!jobQueryValues) {
+    return ''
+  }
+  let query = `cluster=${jobQueryValues.clusterName}`
+
+  const trimmedUsernames = jobQueryValues.usernames || ''
+
+  const userNameList = trimmedUsernames ? trimmedUsernames.split(',').map(item => item.trim()) : []
+  if (userNameList?.length === 0) {
```

Comment:
I think the question mark (null check operator) should not be needed here or below, since userNameList is either the result of map (hence an array) or it is the empty array?